### PR TITLE
Udi update repo comment seperate comments

### PR DESCRIPTION
**Pull request by udiraz2**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](https://github.com/checkmarx-ts/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/checkmarx-ltd/open-source-template/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

> Describe the purpose of this PR along with any background information and the impacts of the proposed change.

### References

> Include supporting link to GitHub Issue/PR number

### Testing

> Describe how this change was tested. Be specific about anything not tested and reasons why. If this solution has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentaiton is a Wiki Update, please indicate desired changes within PR MD Comment*
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

**Comment by cxflowtestuser** [comment]
### Checkmarx scan completed
[Full Scan Details](http://EC2AMAZ-JP2I9QF/CxWebClient/ViewerMain.aspx?scanid=1000123&projectid=51)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|549
Low|5771
Informational|44

#### Violation Summary
Severity|Count
---|---

#### Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---

File: src/main/java/com/checkmarx/flow/service/GitHubService.java [hunk 1]
```diff
@@ -82,50 +84,66 @@ void processPull(ScanRequest request, ScanResults results) throws GitHubClientEx
...
+                if (!PullRequestCommentsHelper.shouldUpdateComment(comment, commentToUpdate.getComment())) {
+                    log.debug("Comment should not be updated");
+                    return;
+                }
+                log.debug("Going to update GitHub pull request comment");
+                updateComment(commentToUpdate.getCommentUrl(), comment);
+            } else {
+                log.debug("Going to create a new GitHub pull request comment");
+                addComment(request, comment);
+            }
```
Comment (AvivCX) [inline src/main/java/com/checkmarx/flow/service/GitHubService.java:146]: need to make sure exception in write comment attempt would is caught and not terminate the entire pull request update process

File: src/main/java/com/checkmarx/flow/service/GitHubService.java [hunk 2]
```diff
@@ -82,50 +84,66 @@ void processPull(ScanRequest request, ScanResults results) throws GitHubClientEx
...
         try {
-            List<RepoComment> repoComments = getComments(request.getMergeNoteUri());
-            log.debug("There are {} checkmarx comments on this pull request", repoComments.size());
-            addComment(request, comment);
+            RepoComment commentToUpdate = PullRequestCommentsHelper.getCommentToUpdate(getComments(request.getMergeNoteUri()), comment);
+            if (commentToUpdate !=  null) {
+                if (!PullRequestCommentsHelper.shouldUpdateComment(comment, commentToUpdate.getComment())) {
+                    log.debug("Comment should not be updated");
+                    return;
+                }
```
Comment (AvivCX) [inline src/main/java/com/checkmarx/flow/service/GitHubService.java:140]: I would add more info logging, especially in the decision-making points, to be able to follow the main flow with info level logs

File: src/main/java/com/checkmarx/flow/service/GitHubService.java [hunk 3]
```diff
@@ -82,50 +84,66 @@ void processPull(ScanRequest request, ScanResults results) throws GitHubClientEx
...
 
     public void sendMergeComment(ScanRequest request, String comment) throws GitHubClientException {
         try {
-            List<RepoComment> repoComments = getComments(request.getMergeNoteUri());
-            log.debug("There are {} checkmarx comments on this pull request", repoComments.size());
-            addComment(request, comment);
+            RepoComment commentToUpdate = PullRequestCommentsHelper.getCommentToUpdate(getComments(request.getMergeNoteUri()), comment);
+            if (commentToUpdate !=  null) {
+                if (!PullRequestCommentsHelper.shouldUpdateComment(comment, commentToUpdate.getComment())) {
+                    log.debug("Comment should not be updated");
```
Comment (AvivCX) [inline src/main/java/com/checkmarx/flow/service/GitHubService.java:138]: I would add more info logging, especially in the decision-making points, to be able to follow the main flow with info level logs

File: src/main/java/com/checkmarx/flow/service/GitHubService.java (hunk 1 above)
Comment (AvivCX) [inline src/main/java/com/checkmarx/flow/service/GitHubService.java:146]: need to make sure exception in write comment attempt would is caught and not terminate the entire pull request update process

**Pull request closed without merging**